### PR TITLE
RDK-45706: Support for hibernate and wakeup Phase 1

### PR DESCRIPTION
**Pull request by adrianM27**
### Description
Add support for container hibernation and wakeup using Memcr tool as a backend

### Test Procedure
Call:
DobbyTool hibernate <container_id>
DobbyTool wakeup <container_id>

### Type of Change
- [ ] New feature (non-breaking change which adds functionality)

### Requires Bitbake Recipe changes?
- [ ] The base Bitbake recipe (`meta-rdk-ext/recipes-containers/dobby/dobby.bb`) must be modified to support the changes in this PR (beyond updating `SRC_REV`)
- [ ] CMake flag has to be set to enable Memcr backend interface implementation: DOBBY_HIBERNATE_MEMCR_IMPL=ON

### CODE REVIEW

File: client/lib/source/DobbyProxy.cpp [hunk 1]
```diff
@@ -93,6 +93,14 @@ DobbyProxy::DobbyProxy(const std::shared_ptr<AI_IPC::IIpcService>& ipcService,
     const AI_IPC::SignalHandler stoppedHandler(std::bind(&DobbyProxy::onContainerStoppedEvent, this, std::placeholders::_1));
     mContainerStoppedSignal = mIpcService->registerSignalHandler(stoppedSignal, stoppedHandler);
 
+    const AI_IPC::Signal hibernatedSignal(objectName, DOBBY_CTRL_INTERFACE, DOBBY_CTRL_EVENT_HIBERNATED);
+    const AI_IPC::SignalHandler hibernatedHandler(std::bind(&DobbyProxy::onContainerHibernatedEvent, this, std::placeholders::_1));
+    mContainerStartedSignal = mIpcService->registerSignalHandler(hibernatedSignal, hibernatedHandler);
+
+    const AI_IPC::Signal awokenSignal(objectName, DOBBY_CTRL_INTERFACE, DOBBY_CTRL_EVENT_AWOKEN);
+    const AI_IPC::SignalHandler awokenHandler(std::bind(&DobbyProxy::onContainerAwokenEvent, this, std::placeholders::_1));
+    mContainerStartedSignal = mIpcService->registerSignalHandler(hibernatedSignal, hibernatedHandler);
```
Comment (goruklu) [inline client/lib/source/DobbyProxy.cpp:102]: shouldn't this be

`mIpcService->registerSignalHandler(awokenSignal, awokenHandler);`

**Comment by goruklu** [comment]
@adrianM27 some CI verification builds are failing, please check. Also, it might be a good idea to add the DOBBY_HIBERNATE_MEMCR_IMPL=ON option to the build job in github workflow. Without it, the CI build jobs will not enable the hibernate feature.

File: daemon/lib/source/DobbyManager.cpp [hunk 2]
```diff
@@ -1514,6 +1522,240 @@ bool DobbyManager::resumeContainer(int32_t cd)
...
+                delayMs = std::stoi(&options[delayMsPos + std::string("--delay=").length()], nullptr, 10);
+            }
+            DobbyHibernate::Error ret = DobbyHibernate::Error::ErrorNone;
+
+            int delayChunkMs = 100;
+            while (delayMs > 0)
+            {
+                int sleepTime = delayMs > delayChunkMs? delayChunkMs : delayMs;
+                delayMs -= sleepTime;
+                std::this_thread::sleep_for(std::chrono::milliseconds(sleepTime));
```
[the remark this answers is not in the document]
Comment (adrianM27) [reply daemon/lib/source/DobbyManager.cpp:1587]: Yes, both, and especially to abort this thread then 'wakeup' was called

File: client/lib/source/DobbyProxy.cpp (hunk 1 above)
Comment (adrianM27) [reply client/lib/source/DobbyProxy.cpp:102]: yes, my bad, fixed

**Review by goruklu — APPROVED** [review]

**Pull request merged**